### ISSUE TITLE
Problems linking a library. undefined reference 

### ISSUE THREAD

**Issue by caballero03** [AlexandreRouma/SDRPlusPlus#891]
Hello,

I am writing a custom module for SDR++ for my own use. This module will need to access a I2C bus on a Raspberry Pi. 

I have gotten the module to compile and it works fine with the application, it shows in the menu and even get retune events properly. However, when I tried to fold in the I2C stuff using libi2c, it compiles but won't link the I2C library and complains with: undefined reference to `i2c_smbus_read_word_data'

I have included the headers with:
extern "C" {
#include <linux/i2c-dev.h>
#include <i2c/smbus.h>
}

I have added the library in the CMakeLists.txt for my module many different ways:
I thought this would work:
target_link_libraries(hf_band_decoder PRIVATE sdrpp_core i2c)

I even tried this:
target_link_libraries(hf_band_decoder PRIVATE sdrpp_core /usr/lib/aarch64-linux-gnu/libi2c.so)

When I do get it to build, with the I2C code commented out, and run `ldd sdrpp` it does not show libi2c as a library. So it would appear that it is not getting link in the end.

I hope someone can lead me down a better path. I have tried everything I can find to fix this but nothing seems to work. Is there something that I am missing? Some kind of global link declaration file or something? Any ideas you may have would be very helpful.

Thank you for you time. I hope I have hair left after this. :-)

**Comment by caballero03** [comment]
Here is a little more detail.

I added `set(CMAKE_VERBOSE_MAKEFILE ON)` to my CMakeLists.txt file and got additional info such follows:

`/usr/bin/c++ -fPIC -g -shared -Wl,-soname,hf_band_decoder.so -o hf_band_decoder.so CMakeFiles/hf_band_decoder.dir/src/PreselectorControl.cpp.o CMakeFiles/hf_band_decoder.dir/src/main.cpp.o  -Wl,-rpath,/tmp/tmp.b0DyTIevd4/cmake-build-debug/core: ../../core/libsdrpp_core.so /usr/lib/aarch64-linux-gnu/libi2c.so ../../core/libcorrect/lib/libcorrect.a -lglfw /usr/lib/aarch64-linux-gnu/libOpenGL.so /usr/lib/aarch64-linux-gnu/libGLX.so /usr/lib/aarch64-linux-gnu/libGLU.so -lfftw3f -lglfw /usr/lib/aarch64-linux-gnu/libOpenGL.so /usr/lib/aarch64-linux-gnu/libGLX.so /usr/lib/aarch64-linux-gnu/libGLU.so -lfftw3f -lvolk -lzstd -lstdc++fs`

I would appear that the I2C library is getting to the compiler options.

I am pretty new with CMake. I'm sure I am just missing something simple. Thanks for reading.

**Comment by caballero03** [comment]
Another update on my issue. I gave up fighting with the library. There is clearly something wrong with my version or something.

I discovered that the libi2c library is just a thin wrapper around some sys/ioctl calls to the i2c device file. I found the source code for the smbus wrapper and copied the functions that I wanted into my own .cpp file and Bob's your uncle! Now I don't need any external library. While this may not necessarily be the best way to go about this, I think it'll be fine for what I am doing.

Thank you to any and all who may have tried to figure out this issue. Every solution I found didn't seem to work in my case.
